superset-frontend/src/explore/components/DatasourcePanel.tsx

```diff
@@ -109,45 +111,41 @@ export default function DataSourcePanel({
     metrics,
   });
 
+  function searchByRelevance(
+    datasource: Array<ColumnMeta | Metric>,
+    value: string,
+    isMetrics = false,
+  ) {
+    const properties = [
+      isMetrics ? 'metric_name' : 'column_name',
+      'description',
+      'verbose_name',
```

Comment:
`verbose_name` should rank first because it is what's been displayed to the end user.

I would also suggest adding back the different rankings since as it's very unlikely people would search description or SQL expressions by acronyms.

Reply:
the problem with this is that columns with description containing the searched keyword will go to the top, which is not what we want @junlincc could you chime in on this?

Reply:
what's been displayed to the end user is` metric_name` or `column name`. `verbose_name` is auto generated and show by default when these two are missing, kinda random we believe? or can you explain how `verbose_name` is generated and how to change it in the UI. @ktmud 
and yes, we don't want columns with description containing the searched keyword goes to the top. @nikolagigic

Reply:
<img width="936" alt="verbose-name" src="https://user-images.githubusercontent.com/335541/105932285-8e166c00-6001-11eb-8aae-d594f42c5672.png">

`verbose_name` is the metric/column label when specified. It take precedent of `metric_name` and `column_name` in the UI. The logic is, if `verbose_name` is set, use `verbose_name`, otherwise fallback to `metric_name` or `column_name`. Funny thing is, when `verbose_name` is set, there is actually no way in the UI to view what the underlying `metric_name` and `column_name` is. We may or may not want to do something about this.

Reply:
Changing the ranking of `description` and `expression` from the default `MATCHES` to `CONTAINS` is a separate issue from the ranking across properties.

After reading more closely on [`match-sorter`'s doc](https://www.npmjs.com/package/match-sorter), I think you can achieve the same thing (rank `description` and `expression` lower) by utilizing the `minRanking` option.

Changes based on my original code:

```ts
     setList({
      columns: matchSorter(columns, value, {
        keys: [
          {
            key: 'verbose_name',
            minRanking: rankings.EQUAL,
          },
          {
            key: 'column_name',
            minRanking: rankings.EQUAL,
          },
          {
            key: 'description',
            threshold: rankings.CONTAINS,
          },
          {
            key: 'expression',
            threshold: rankings.CONTAINS,
          },
        ],
        keepDiacritics: true,
      }),
      metrics: matchSorter(metrics, value, {
        keys: [
          {
            key: 'verbose_name',
            minRanking: rankings.EQUAL,
          },
          {
            key: 'metric_name',
            minRanking: rankings.EQUAL,
          },
          {
            key: 'description',
            threshold: rankings.CONTAINS,
          },
          {
            key: 'expression',
            threshold: rankings.CONTAINS,
          },
        ],
        keepDiacritics: true,
        baseSort: (a, b) =>
          Number(b.item.is_certified) - Number(a.item.is_certified) ||
          String(a.rankedValue).localeCompare(b.rankedValue),
      }),
    });
```

Reply:
<img src="https://user-images.githubusercontent.com/335541/105934799-b4d6a180-6005-11eb-93e5-997c59cc45b5.png" width="300">

Can confirm this solution works.

Reply:
using **EQUAL** for verbose_name would still rank some of the columns with the acronym of a keyword higher. 
![Screenshot 2021-01-27 at 06 33 28](https://user-images.githubusercontent.com/26679866/105947799-0dd82100-606a-11eb-87e1-a53f635ce363.png)

Howver using **ACRONYM** instead would solve the issue imo
![Screenshot 2021-01-27 at 06 35 10](https://user-images.githubusercontent.com/26679866/105947833-234d4b00-606a-11eb-80dc-987628c445be.png)

Reply:
@ktmud if we can resolve above ^, I agree `verbose_name` should rank higher. clearly i still have a lot to learn in Superset.

Reply:
@nikolagigic to the end users, verbose names and metric/column names are almost indistinguishable in the list (except casing difference enforced only by convention), so I'm not sure ranking verbose names lower make that much of a difference.

I also noticed two other problems:

1. When using `minRanking`, items are promoted to the top as long as there is a simple match (e.g. `evt` will match `event` because letters occur in the same order), which IMO is not very useful for most cases. For example, in your screenshots, `end` probably shouldn't match `codeevnt_xxxx`. I don't consider this a high priority bug, but if we do want to fix it, the cleanest solution is probably still the multi-step search you originally did in this PR. Sorry for the back and forth.
2. Fields with new lines are not fully searchable. For example, "Highest Degree Earned" has SQL expression:
    ```
    CASE
        WHEN ....
    ```
    but searching for "case when" will not produce a match.

   We may want to do something like this:

   ```ts
     columns: matchSorter(columns, value, {
        keys: [
          {
            key: item =>
              [item.verbose_name, item.column_name].map(
                x => x?.replace(/[_\n\s]+/g, ' ') || '',
              ),
            minRanking: rankings.EQUAL,
          },
          {
            key: item =>
              [item.description, item.expression].map(
                x => x?.replace(/[_\n\s]+/g, ' ') || '',
              ),
            threshold: rankings.CONTAINS,
          },
        ],
        keepDiacritics: true,
      }),
   ```